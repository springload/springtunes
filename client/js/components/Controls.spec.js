import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import { shallow } from 'enzyme';
import Controls from './Controls';

const emptyFunc = () => {};

describe('Controls component', () => {
    it('renders correctly with min params', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        const result = shallowRenderer.render(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={false}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );
        expect(result).toMatchSnapshot();
    });

    it('renders correctly with isFetching true', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        const result = shallowRenderer.render(
            <Controls
                isFetching={true}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={false}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );
        expect(result).toMatchSnapshot();
    });

    it('renders correctly with isModifyingMute true', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        const result = shallowRenderer.render(
            <Controls
                isFetching={false}
                isModifyingMute={true}
                isMuted={false}
                isPlaying={false}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );
        expect(result).toMatchSnapshot();
    });

    it('renders correctly with isMuted', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        const result = shallowRenderer.render(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={true}
                isPlaying={false}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );
        expect(result).toMatchSnapshot();
    });

    it('renders correctly with isPlaying true', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        const result = shallowRenderer.render(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={true}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );
        expect(result).toMatchSnapshot();
    });

    it('execute correctly refreshClick', () => {
        let testClick = false;
        const testOnClick = () => { testClick = true; };

        const component = shallow(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={true}
                volume={50}
                refreshClick={testOnClick}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );

        component.find('button').first().simulate('click');
        expect(testClick).toBeTruthy();
    });

    it('execute correctly backClick', () => {
        let testClick = false;
        const testOnClick = () => { testClick = true; };

        const component = shallow(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={true}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={testOnClick}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );

        component.find('button').at(1).simulate('click');
        expect(testClick).toBeTruthy();
    });

    it('execute correctly togglePauseClick', () => {
        let testClick = false;
        const testOnClick = () => { testClick = true; };

        const component = shallow(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={true}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={testOnClick}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );

        component.find('button').at(2).simulate('click');
        expect(testClick).toBeTruthy();
    });

    it('execute correctly nextClick', () => {
        let testClick = false;
        const testOnClick = () => { testClick = true; };

        const component = shallow(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={true}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={testOnClick}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );

        component.find('button').at(3).simulate('click');
        expect(testClick).toBeTruthy();
    });

    it('execute correctly muteClick', () => {
        let testClick = false;
        const testOnClick = () => { testClick = true; };

        const component = shallow(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={true}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={testOnClick}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );

        component.find('button').at(4).simulate('click');
        expect(testClick).toBeTruthy();
    });

    it('execute correctly unmuteClick', () => {
        let testClick = false;
        const testOnClick = () => { testClick = true; };

        const component = shallow(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={true}
                isPlaying={true}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={testOnClick}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );

        component.find('button').at(4).simulate('click');
        expect(testClick).toBeTruthy();
    });

    // should show all the buttons when not muted
    it('renders correctly numbers of buttons', () => {
        const component = shallow(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={false}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );

        expect(component.find('button').length).toBe(6);
    });

    // should show all the buttons when muted
    it('renders correctly numbers of buttons', () => {
        const component = shallow(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={true}
                isPlaying={false}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );

        expect(component.find('button').length).toBe(6);
    });

    it('updateURL works', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        shallowRenderer.render(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={false}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );
        const instance = shallowRenderer.getMountedInstance();
        instance.updateURL('http://newurl');
        expect(instance.state.url).toBe('http://newurl');
    });

    it('checkURL works invalid', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        shallowRenderer.render(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={false}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );
        const instance = shallowRenderer.getMountedInstance();
        instance.checkURL('http://newurl');
        expect(instance.state.error).toBe('Invalid URL provided. You can only play playlists. URL must start with \'https://open.spotify.com/user/...\'');
    });

    it('checkURL works empty', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        shallowRenderer.render(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={false}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );
        const instance = shallowRenderer.getMountedInstance();
        instance.checkURL('');
        expect(instance.state.error).toBe('Invalid URL provided. You can only play playlists. URL must start with \'https://open.spotify.com/user/...\'');
    });

    it('checkURL works valid', () => {
        const shallowRenderer = ReactTestUtils.createRenderer();
        shallowRenderer.render(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={false}
                isPlaying={false}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );
        const instance = shallowRenderer.getMountedInstance();
        instance.checkURL('https://open.spotify.com/user/springload/playlist/6oYWzKyhPbfy2I83KS7JK5');
        expect(instance.state.error).toBe('');
    });

    it('submit form', () => {
        let testClick = false;
        const testOnClick = () => { testClick = true; };

        const component = shallow(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={true}
                isPlaying={true}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={testOnClick}
            />,
        );

        component.find('form').at(0).simulate('submit', { preventDefault: () => {} });
        expect(testClick).toBeFalsy();

        expect(component).toMatchSnapshot();
    });

    it('input change', () => {
        const component = shallow(
            <Controls
                isFetching={false}
                isModifyingMute={false}
                isMuted={true}
                isPlaying={true}
                volume={50}
                refreshClick={emptyFunc}
                togglePauseClick={emptyFunc}
                backClick={emptyFunc}
                nextClick={emptyFunc}
                muteClick={emptyFunc}
                unmuteClick={emptyFunc}
                volumeChange={emptyFunc}
                playURL={emptyFunc}
            />,
        );

        component.find('input').at(0).simulate('change', { target: { value: 'newValue' } });
        expect(component.state().url).toBe('newValue');
    });
});
