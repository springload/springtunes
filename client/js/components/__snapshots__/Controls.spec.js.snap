exports[`Controls component renders correctly with isFetching true 1`] = `
<div
  className="controls">
  <button
    className="btn-orange"
    disabled={true}
    onClick={[Function]}>
    <Icon
      iconType="small"
      name="fa-refresh" />
     Refresh
  </button>
  <button
    disabled={true}
    onClick={[Function]}>
    <Icon
      iconType="small"
      name="fa-step-backward" />
     Back
  </button>
  <button
    disabled={true}
    onClick={[Function]}>
    <span>
      <Icon
        iconType="small"
        name="fa-play" />
       Play
    </span>
  </button>
  <button
    disabled={true}
    onClick={[Function]}>
    Next 
    <Icon
      iconType="small"
      name="fa-step-forward" />
  </button>
  <button
    disabled={false}
    onClick={[Function]}
    title="Mute">
    <Icon
      iconType="small"
      name="fa-volume-up" />
  </button>
  <div
    className="volume-control">
    <span>
      Volume
    </span>
    <Slider
      allowCross={true}
      className=""
      disabled={true}
      dots={false}
      handle={<Handle />}
      included={true}
      marks={Object {}}
      max={100}
      min={0}
      onAfterChange={[Function]}
      onBeforeChange={[Function]}
      onChange={[Function]}
      prefixCls="rc-slider"
      range={false}
      step={1}
      tipFormatter={[Function]}
      tipTransitionName=""
      value={50}
      vertical={false} />
  </div>
  <div
    className="url-control">
    <form
      onSubmit={[Function]}>
      <label
        className="input__label"
        htmlFor="spotify-url">
        <span
          className="u-accessible">
          Spotify URL
        </span>
        <input
          className="input--large"
          id="spotify-url"
          name="url"
          onChange={[Function]}
          placeholder="https://open.spotify.com/user/spotify/playlist/..."
          type="text" />
      </label>
      <button
        disabled={true}
        title="Play URL">
        Play URL
      </button>
    </form>
  </div>
</div>
`;

exports[`Controls component renders correctly with isModifyingMute true 1`] = `
<div
  className="controls">
  <button
    className="btn-orange"
    disabled={false}
    onClick={[Function]}>
    <Icon
      iconType="small"
      name="fa-refresh" />
     Refresh
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    <Icon
      iconType="small"
      name="fa-step-backward" />
     Back
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    <span>
      <Icon
        iconType="small"
        name="fa-play" />
       Play
    </span>
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    Next 
    <Icon
      iconType="small"
      name="fa-step-forward" />
  </button>
  <button
    disabled={true}
    onClick={[Function]}
    title="Mute">
    <Icon
      iconType="small"
      name="fa-volume-up" />
  </button>
  <div
    className="volume-control">
    <span>
      Volume
    </span>
    <Slider
      allowCross={true}
      className=""
      disabled={false}
      dots={false}
      handle={<Handle />}
      included={true}
      marks={Object {}}
      max={100}
      min={0}
      onAfterChange={[Function]}
      onBeforeChange={[Function]}
      onChange={[Function]}
      prefixCls="rc-slider"
      range={false}
      step={1}
      tipFormatter={[Function]}
      tipTransitionName=""
      value={50}
      vertical={false} />
  </div>
  <div
    className="url-control">
    <form
      onSubmit={[Function]}>
      <label
        className="input__label"
        htmlFor="spotify-url">
        <span
          className="u-accessible">
          Spotify URL
        </span>
        <input
          className="input--large"
          id="spotify-url"
          name="url"
          onChange={[Function]}
          placeholder="https://open.spotify.com/user/spotify/playlist/..."
          type="text" />
      </label>
      <button
        disabled={false}
        title="Play URL">
        Play URL
      </button>
    </form>
  </div>
</div>
`;

exports[`Controls component renders correctly with isMuted 1`] = `
<div
  className="controls">
  <button
    className="btn-orange"
    disabled={false}
    onClick={[Function]}>
    <Icon
      iconType="small"
      name="fa-refresh" />
     Refresh
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    <Icon
      iconType="small"
      name="fa-step-backward" />
     Back
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    <span>
      <Icon
        iconType="small"
        name="fa-play" />
       Play
    </span>
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    Next 
    <Icon
      iconType="small"
      name="fa-step-forward" />
  </button>
  <button
    disabled={false}
    onClick={[Function]}
    title="Unmute">
    <Icon
      iconType="small"
      name="fa-volume-off" />
  </button>
  <div
    className="volume-control">
    <span>
      Volume
    </span>
    <Slider
      allowCross={true}
      className=""
      disabled={false}
      dots={false}
      handle={<Handle />}
      included={true}
      marks={Object {}}
      max={100}
      min={0}
      onAfterChange={[Function]}
      onBeforeChange={[Function]}
      onChange={[Function]}
      prefixCls="rc-slider"
      range={false}
      step={1}
      tipFormatter={[Function]}
      tipTransitionName=""
      value={50}
      vertical={false} />
  </div>
  <div
    className="url-control">
    <form
      onSubmit={[Function]}>
      <label
        className="input__label"
        htmlFor="spotify-url">
        <span
          className="u-accessible">
          Spotify URL
        </span>
        <input
          className="input--large"
          id="spotify-url"
          name="url"
          onChange={[Function]}
          placeholder="https://open.spotify.com/user/spotify/playlist/..."
          type="text" />
      </label>
      <button
        disabled={false}
        title="Play URL">
        Play URL
      </button>
    </form>
  </div>
</div>
`;

exports[`Controls component renders correctly with isPlaying true 1`] = `
<div
  className="controls">
  <button
    className="btn-orange"
    disabled={false}
    onClick={[Function]}>
    <Icon
      iconType="small"
      name="fa-refresh" />
     Refresh
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    <Icon
      iconType="small"
      name="fa-step-backward" />
     Back
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    <span>
      <Icon
        iconType="small"
        name="fa-pause" />
       Pause
    </span>
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    Next 
    <Icon
      iconType="small"
      name="fa-step-forward" />
  </button>
  <button
    disabled={false}
    onClick={[Function]}
    title="Mute">
    <Icon
      iconType="small"
      name="fa-volume-up" />
  </button>
  <div
    className="volume-control">
    <span>
      Volume
    </span>
    <Slider
      allowCross={true}
      className=""
      disabled={false}
      dots={false}
      handle={<Handle />}
      included={true}
      marks={Object {}}
      max={100}
      min={0}
      onAfterChange={[Function]}
      onBeforeChange={[Function]}
      onChange={[Function]}
      prefixCls="rc-slider"
      range={false}
      step={1}
      tipFormatter={[Function]}
      tipTransitionName=""
      value={50}
      vertical={false} />
  </div>
  <div
    className="url-control">
    <form
      onSubmit={[Function]}>
      <label
        className="input__label"
        htmlFor="spotify-url">
        <span
          className="u-accessible">
          Spotify URL
        </span>
        <input
          className="input--large"
          id="spotify-url"
          name="url"
          onChange={[Function]}
          placeholder="https://open.spotify.com/user/spotify/playlist/..."
          type="text" />
      </label>
      <button
        disabled={false}
        title="Play URL">
        Play URL
      </button>
    </form>
  </div>
</div>
`;

exports[`Controls component renders correctly with min params 1`] = `
<div
  className="controls">
  <button
    className="btn-orange"
    disabled={false}
    onClick={[Function]}>
    <Icon
      iconType="small"
      name="fa-refresh" />
     Refresh
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    <Icon
      iconType="small"
      name="fa-step-backward" />
     Back
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    <span>
      <Icon
        iconType="small"
        name="fa-play" />
       Play
    </span>
  </button>
  <button
    disabled={false}
    onClick={[Function]}>
    Next 
    <Icon
      iconType="small"
      name="fa-step-forward" />
  </button>
  <button
    disabled={false}
    onClick={[Function]}
    title="Mute">
    <Icon
      iconType="small"
      name="fa-volume-up" />
  </button>
  <div
    className="volume-control">
    <span>
      Volume
    </span>
    <Slider
      allowCross={true}
      className=""
      disabled={false}
      dots={false}
      handle={<Handle />}
      included={true}
      marks={Object {}}
      max={100}
      min={0}
      onAfterChange={[Function]}
      onBeforeChange={[Function]}
      onChange={[Function]}
      prefixCls="rc-slider"
      range={false}
      step={1}
      tipFormatter={[Function]}
      tipTransitionName=""
      value={50}
      vertical={false} />
  </div>
  <div
    className="url-control">
    <form
      onSubmit={[Function]}>
      <label
        className="input__label"
        htmlFor="spotify-url">
        <span
          className="u-accessible">
          Spotify URL
        </span>
        <input
          className="input--large"
          id="spotify-url"
          name="url"
          onChange={[Function]}
          placeholder="https://open.spotify.com/user/spotify/playlist/..."
          type="text" />
      </label>
      <button
        disabled={false}
        title="Play URL">
        Play URL
      </button>
    </form>
  </div>
</div>
`;

exports[`Controls component submit form 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="controls">
    <button
        className="btn-orange"
        disabled={false}
        onClick={[Function]}>
        <Icon
            iconType="small"
            name="fa-refresh" />
         Refresh
    </button>
    <button
        disabled={false}
        onClick={[Function]}>
        <Icon
            iconType="small"
            name="fa-step-backward" />
         Back
    </button>
    <button
        disabled={false}
        onClick={[Function]}>
        <span>
            <Icon
                iconType="small"
                name="fa-pause" />
             Pause
        </span>
    </button>
    <button
        disabled={false}
        onClick={[Function]}>
        Next 
        <Icon
            iconType="small"
            name="fa-step-forward" />
    </button>
    <button
        disabled={false}
        onClick={[Function]}
        title="Unmute">
        <Icon
            iconType="small"
            name="fa-volume-off" />
    </button>
    <div
        className="volume-control">
        <span>
            Volume
        </span>
        <Slider
            allowCross={true}
            className=""
            disabled={false}
            dots={false}
            handle={<Handle />}
            included={true}
            marks={Object {}}
            max={100}
            min={0}
            onAfterChange={[Function]}
            onBeforeChange={[Function]}
            onChange={[Function]}
            prefixCls="rc-slider"
            range={false}
            step={1}
            tipFormatter={[Function]}
            tipTransitionName=""
            value={50}
            vertical={false} />
    </div>
    <div
        className="url-control">
        <form
            onSubmit={[Function]}>
            <label
                className="input__label"
                htmlFor="spotify-url">
                <span
                    className="u-accessible">
                    Spotify URL
                </span>
                <input
                    className="input--large"
                    id="spotify-url"
                    name="url"
                    onChange={[Function]}
                    placeholder="https://open.spotify.com/user/spotify/playlist/..."
                    type="text" />
            </label>
            <button
                disabled={false}
                title="Play URL">
                Play URL
            </button>
            <div
                className="error">
                Invalid URL provided. You can only play playlists. URL must start with \'https://open.spotify.com/user/...\'
            </div>
        </form>
    </div>
</div>,
  "nodes": Array [
    <div
      className="controls">
      <button
            className="btn-orange"
            disabled={false}
            onClick={[Function]}>
            <Icon
                  iconType="small"
                  name="fa-refresh" />
             Refresh
      </button>
      <button
            disabled={false}
            onClick={[Function]}>
            <Icon
                  iconType="small"
                  name="fa-step-backward" />
             Back
      </button>
      <button
            disabled={false}
            onClick={[Function]}>
            <span>
                  <Icon
                        iconType="small"
                        name="fa-pause" />
                   Pause
            </span>
      </button>
      <button
            disabled={false}
            onClick={[Function]}>
            Next 
            <Icon
                  iconType="small"
                  name="fa-step-forward" />
      </button>
      <button
            disabled={false}
            onClick={[Function]}
            title="Unmute">
            <Icon
                  iconType="small"
                  name="fa-volume-off" />
      </button>
      <div
            className="volume-control">
            <span>
                  Volume
            </span>
            <Slider
                  allowCross={true}
                  className=""
                  disabled={false}
                  dots={false}
                  handle={<Handle />}
                  included={true}
                  marks={Object {}}
                  max={100}
                  min={0}
                  onAfterChange={[Function]}
                  onBeforeChange={[Function]}
                  onChange={[Function]}
                  prefixCls="rc-slider"
                  range={false}
                  step={1}
                  tipFormatter={[Function]}
                  tipTransitionName=""
                  value={50}
                  vertical={false} />
      </div>
      <div
            className="url-control">
            <form
                  onSubmit={[Function]}>
                  <label
                        className="input__label"
                        htmlFor="spotify-url">
                        <span
                              className="u-accessible">
                              Spotify URL
                        </span>
                        <input
                              className="input--large"
                              id="spotify-url"
                              name="url"
                              onChange={[Function]}
                              placeholder="https://open.spotify.com/user/spotify/playlist/..."
                              type="text" />
                  </label>
                  <button
                        disabled={false}
                        title="Play URL">
                        Play URL
                  </button>
                  <div
                        className="error">
                        Invalid URL provided. You can only play playlists. URL must start with \'https://open.spotify.com/user/...\'
                  </div>
            </form>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Controls
        backClick={[Function]}
        isFetching={false}
        isModifyingMute={false}
        isMuted={true}
        isPlaying={true}
        muteClick={[Function]}
        nextClick={[Function]}
        playURL={[Function]}
        refreshClick={[Function]}
        togglePauseClick={[Function]}
        unmuteClick={[Function]}
        volume={50}
        volumeChange={[Function]} />,
      "_debugID": 35,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Controls {
        "_reactInternalInstance": [Circular],
        "checkURL": [Function],
        "context": Object {},
        "props": Object {
          "backClick": [Function],
          "isFetching": false,
          "isModifyingMute": false,
          "isMuted": true,
          "isPlaying": true,
          "muteClick": [Function],
          "nextClick": [Function],
          "playURL": [Function],
          "refreshClick": [Function],
          "togglePauseClick": [Function],
          "unmuteClick": [Function],
          "volume": 50,
          "volumeChange": [Function],
        },
        "refs": Object {},
        "state": Object {
          "error": "Invalid URL provided. You can only play playlists. URL must start with \'https://open.spotify.com/user/...\'",
          "url": "",
        },
        "updateURL": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 18,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="controls">
          <button
                    className="btn-orange"
                    disabled={false}
                    onClick={[Function]}>
                    <Icon
                              iconType="small"
                              name="fa-refresh" />
                     Refresh
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}>
                    <Icon
                              iconType="small"
                              name="fa-step-backward" />
                     Back
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}>
                    <span>
                              <Icon
                                        iconType="small"
                                        name="fa-pause" />
                               Pause
                    </span>
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}>
                    Next 
                    <Icon
                              iconType="small"
                              name="fa-step-forward" />
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}
                    title="Unmute">
                    <Icon
                              iconType="small"
                              name="fa-volume-off" />
          </button>
          <div
                    className="volume-control">
                    <span>
                              Volume
                    </span>
                    <Slider
                              allowCross={true}
                              className=""
                              disabled={false}
                              dots={false}
                              handle={<Handle />}
                              included={true}
                              marks={Object {}}
                              max={100}
                              min={0}
                              onAfterChange={[Function]}
                              onBeforeChange={[Function]}
                              onChange={[Function]}
                              prefixCls="rc-slider"
                              range={false}
                              step={1}
                              tipFormatter={[Function]}
                              tipTransitionName=""
                              value={50}
                              vertical={false} />
          </div>
          <div
                    className="url-control">
                    <form
                              onSubmit={[Function]}>
                              <label
                                        className="input__label"
                                        htmlFor="spotify-url">
                                        <span
                                                  className="u-accessible">
                                                  Spotify URL
                                        </span>
                                        <input
                                                  className="input--large"
                                                  id="spotify-url"
                                                  name="url"
                                                  onChange={[Function]}
                                                  placeholder="https://open.spotify.com/user/spotify/playlist/..."
                                                  type="text" />
                              </label>
                              <button
                                        disabled={false}
                                        title="Play URL">
                                        Play URL
                              </button>
                              <div
                                        className="error">
                                        Invalid URL provided. You can only play playlists. URL must start with \'https://open.spotify.com/user/...\'
                              </div>
                    </form>
          </div>
</div>,
        "_debugID": 36,
        "_renderedOutput": <div
          className="controls">
          <button
                    className="btn-orange"
                    disabled={false}
                    onClick={[Function]}>
                    <Icon
                              iconType="small"
                              name="fa-refresh" />
                     Refresh
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}>
                    <Icon
                              iconType="small"
                              name="fa-step-backward" />
                     Back
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}>
                    <span>
                              <Icon
                                        iconType="small"
                                        name="fa-pause" />
                               Pause
                    </span>
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}>
                    Next 
                    <Icon
                              iconType="small"
                              name="fa-step-forward" />
          </button>
          <button
                    disabled={false}
                    onClick={[Function]}
                    title="Unmute">
                    <Icon
                              iconType="small"
                              name="fa-volume-off" />
          </button>
          <div
                    className="volume-control">
                    <span>
                              Volume
                    </span>
                    <Slider
                              allowCross={true}
                              className=""
                              disabled={false}
                              dots={false}
                              handle={<Handle />}
                              included={true}
                              marks={Object {}}
                              max={100}
                              min={0}
                              onAfterChange={[Function]}
                              onBeforeChange={[Function]}
                              onChange={[Function]}
                              prefixCls="rc-slider"
                              range={false}
                              step={1}
                              tipFormatter={[Function]}
                              tipTransitionName=""
                              value={50}
                              vertical={false} />
          </div>
          <div
                    className="url-control">
                    <form
                              onSubmit={[Function]}>
                              <label
                                        className="input__label"
                                        htmlFor="spotify-url">
                                        <span
                                                  className="u-accessible">
                                                  Spotify URL
                                        </span>
                                        <input
                                                  className="input--large"
                                                  id="spotify-url"
                                                  name="url"
                                                  onChange={[Function]}
                                                  placeholder="https://open.spotify.com/user/spotify/playlist/..."
                                                  type="text" />
                              </label>
                              <button
                                        disabled={false}
                                        title="Play URL">
                                        Play URL
                              </button>
                              <div
                                        className="error">
                                        Invalid URL provided. You can only play playlists. URL must start with \'https://open.spotify.com/user/...\'
                              </div>
                    </form>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Controls
    backClick={[Function]}
    isFetching={false}
    isModifyingMute={false}
    isMuted={true}
    isPlaying={true}
    muteClick={[Function]}
    nextClick={[Function]}
    playURL={[Function]}
    refreshClick={[Function]}
    togglePauseClick={[Function]}
    unmuteClick={[Function]}
    volume={50}
    volumeChange={[Function]} />,
}
`;
